src/data_structures/suffix_array.rs

```diff
@@ -22,6 +22,10 @@ use alphabets::{Alphabet, RankTransform};
 use data_structures::smallints::SmallInts;
 use data_structures::bwt::{DerefBWT, DerefLess, DerefOcc};
 
+// TODO: figure out how to make generic on `SuffixArray` type w/o `as` cast
```

Comment:
@johanneskoester Any idea how to make this work? If I set the `Deref` target to `SuffixArray`, calling any of the functions requires an `as` cast from `Deref<Target = RawSuffixArray>` or `Deref<Target = SampledSuffixArray>` to `Deref<Target = SuffixArray>`. Is this even possible in the current language?